pkg/provisioner/ironic/ironic.go

```diff
@@ -1152,6 +1152,47 @@ func (p *ironicProvisioner) GetFirmwareSettings(includeSchema bool) (settings me
 	return settings, schema, nil
 }
 
+// GetFirmwareComponents gets all available firmware components for a node and return a list.
+func (p *ironicProvisioner) GetFirmwareComponents() ([]metal3api.FirmwareComponentStatus, error) {
+	ironicNode, err := p.getNode()
+	if err != nil {
+		return nil, fmt.Errorf("could not get node to retrieve firmware components: %w", err)
+	}
+
+	// Get the components from Ironic via Gophercloud
+	var componentList []nodes.FirmwareComponent
+	var componentListErr error
+	if p.availableFeatures.HasFirmwareUpdates() {
```

Comment:
if you organizer it as 

```go
if ! p.availableFeatures.HasFirmwareUpdates() {
   return nil, ...
}
```
Then you don't need so many variable declarations.

Reply:
Thanks! I will update